### PR TITLE
feat: [PL-25298]: handle invalid token better

### PR DESCRIPTION
**Pull request by mayankVermaHarness**
##### Summary:

removed "buildLoginUrlFrom401Response" and handled redirection from upstream in `App.tsx`

##### Jira Links:

https://harness.atlassian.net/browse/PL-25298

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/framework/app/App.tsx [hunk 1]
```diff
@@ -90,13 +90,21 @@ export function AppWithAuthentication(props: AppProps): React.ReactElement {
...
       history.push({
         pathname: routes.toRedirect(),
         search: returnUrlParams(getLoginPageURL({ returnUrl: window.location.href }))
       })
     }
+
+    try {
+      const token = SessionToken.getToken()
+      if (!token) {
+        redirectToLoginPage()
```
Comment (abhinavrastogi-harness) [inline src/framework/app/App.tsx:103]: we can use the existing function `global401HandlerUtils()` in  `src/modules/10-common/utils/global401HandlerUtils.tsx`

Comment (mayankVermaHarness) [reply src/framework/app/App.tsx]: done

**Comment by mayankVermaHarness** [comment]
retrigger jest

**Comment by mayankVermaHarness** [comment]
retrigger titlecheck

**Comment by mayankVermaHarness** [comment]
No longer needed. Closing.

**Pull request closed without merging**